symfit/core/minimizers.py

```diff
@@ -64,8 +67,21 @@ class GradientMinimizer(BaseMinimizer):
     def __init__(self, *args, **kwargs):
         jacobian = kwargs.pop('jacobian')
         super(GradientMinimizer, self).__init__(*args, **kwargs)
-        self.jacobian = jacobian
-
+        def _jacobian(*args, **kwargs):
+            # Assume the order of what jacobian returns and parameters is the same
```

Comment:
This assumption is indeed guaranteed, parameters are sorted by their str representation and this is used when deriving the jacobian.